pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -352,6 +362,24 @@ func (ku *Upgrader) upgradeAgentPools(ctx context.Context) error {
 			}
 			ku.logger.Infof("Upgrading Agent VM: %s, pool name: %s", vm.name, *agentPool.Name)
 
+			// copy custom properties from old node to new node if the PreserveNodesProperties in AgentPoolProfile is not set to false explicitly.
+			preserveNodesProperties := true
+			if agentPoolProfile != nil && agentPoolProfile.PreserveNodesProperties != nil {
+				preserveNodesProperties = *agentPoolProfile.PreserveNodesProperties
+			}
+
+			if preserveNodesProperties {
+				if len(newCreatedVMs) > 0 {
+					newNodeName := newCreatedVMs[0]
+					newCreatedVMs = newCreatedVMs[1:]
+					ku.logger.Infof("Copying custom annotations, labels, taints from old node %s to new node %s...", vm.name, newNodeName)
+					err = ku.copyCustomPropertiesToNewNode(client, vm.name, newNodeName)
+					if err != nil {
+						ku.logger.Errorf("Failed to copy custom annotations, labels, taints from old node %s to new node %s: %v", vm.name, newNodeName, err)
```

Comment:
Is this not returning the error on purpose? If we don't want to return an error when we fail to copy node properties, the log should be a warning, not an error.

Reply:
yes, change it to warning.